internal/coreinternal/attraction/coercer.go

```diff
@@ -0,0 +1,69 @@
+package attraction
+
+import (
+	"regexp"
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+var num = regexp.MustCompile(`^(\d+)(?:\.\d+)?$`)
+var dub = regexp.MustCompile(`^(\d+(?:\.\d+)?)$`)
+
+func coerceValue(to string, v pdata.AttributeValue) {
+	switch to {
+	case "string":
+		switch v.Type().String() {
+		case "STRING":
+		default:
+			v.SetStringVal(v.AsString())
+		}
+	case "int":
+		switch v.Type().String() {
+		case "INT":
+		case "DOUBLE":
+			v.SetIntVal(int64(v.DoubleVal()))
+		case "BOOL":
+			if v.BoolVal() {
+				v.SetIntVal(1)
+			} else {
+				v.SetIntVal(0)
+			}
+		case "STRING":
+			s := v.StringVal()
+			n := num.FindStringSubmatch(s)
+			if n != nil {
+				intVal, _ := strconv.Atoi(n[1])
+				v.SetIntVal(int64(intVal))
+			} else {
+				v.SetIntVal(int64(0))
+			}
+		default:
+			v.SetIntVal(int64(0))
```

Comment:
So if the attribute contains unsupported type (e.g. array) it will be actually reset to zero. I am wondering if a safer approach would be to retain the original type and value when conversion cannot be handled

Reply:
That might make sense... my initial rationale was that this should provide a guarantee that the type of the value on the processed attribute was the one requested in the processor action. For the use-cases for which I wanted this it makes no difference however, so I could be convinced either way.

Reply:
Have addressed your other comments, but I'm still not sure on this one... to me it feels like if I specify a convert action, I want a guarantee on the type of the mutated attribute more than I worry about zero-ing a non-convertible field. Will leave it as-is for now, but happy enough to change if there's strong sentiment on it. Of course, the other option would be to add _another_ flag to permit specifying which was the desired behaviour, but that's probably too much...

Reply:
I think it largely depends on the consumer and whether it prefers a valid type over an accurate value. I can imagine either of the cases, though have some preference for the latter. 

When dealing with attribute value, type must be checked anyways and if it's not supported, it should be handled gracefully (e.g. a default value or skip processing), so I think that rarely the unexpected type will cause a lot of issues.

Removing the original value will make tracking the cause the troubleshooting much harder though. E.g. if I'm anticipating string or int in `http.status_code` and got `0` it would be hard to tell if this was due to being sent as non-numeric string, an array or so (or maybe it was originally sent as 0)

Reply:
OK OK I'm sold. Behaviour amended